### PR TITLE
Added token position tests for annotations

### PR DESCRIPTION
**Pull request by TwoOfTwelve**
Added some token position tests for annotations. The tests currently fail, because I decided on different lengths than the java module currently reports.

The lengths I assume for these test:

-Annotation: from @ to end of annotation name
-Annotation type begin: from the first modifier to the end of the name
All other tokens lengths are the same as in the current implementation 

There are also some considerations for the module:

Most of the annotation token types seem to be unused. Maybe we should remove them?
I guess ANNO_M was supposed to be annotation members. They are currently extracted as Method tokens. Do we want to keep it like that?
Annotation arguments are currently not extracted at all. Do we want some tokens for that?

### CODE REVIEW

File: languages/java/src/main/java/de/jplag/java/TokenGeneratingTreeScanner.java [hunk 1]
```diff
@@ -146,7 +147,7 @@ public Void visitClass(ClassTree node, Void unused) {
         } else if (node.getKind() == Tree.Kind.RECORD) {
             addToken(JavaTokenType.J_RECORD_BEGIN, start, 1, semantics);
         } else if (node.getKind() == Tree.Kind.ANNOTATION_TYPE) {
-            addToken(JavaTokenType.J_ANNO_T_BEGIN, start, 10, semantics);
+            addToken(JavaTokenType.J_ANNO_T_BEGIN, start - 2 /* @ is final modifier for annotations */, (start - 2) + 11 + nameLength, semantics);
```
Comment (uuqjz) [inline languages/java/src/main/java/de/jplag/java/TokenGeneratingTreeScanner.java:150]: This inline comment look awful between two parameters

**Review by uuqjz — CHANGES_REQUESTED** [review]
Please move the comment

**Review by tsaglam — APPROVED** [review]
Looks good.

My initial thoughts on the open points:
> Most of the annotation token types seem to be unused. Maybe we should remove them?

Yes, we can do that, probably in a separate PR to also cover #1865.

> I guess ANNO_M was supposed to be annotation members. They are currently extracted as Method tokens. Do we want to keep it like that?

During the declaration of the annotation members in the annotation definition? There I would stick to method tokens.

> Annotation arguments are currently not extracted at all. Do we want some tokens for that?

We could actually do this. However, ideally, it would only be for mandatory ones. If we do it for optional ones, you can affect the tokens by removing the argument or adding an argument with the same value as the default.

Let me know what you guys think!

**Comment by tsaglam** [comment]
@uuqjz ready from your side?

**Review by uuqjz — APPROVED** [review]
LGTM

**Pull request merged**